pages/api/slots.js

```diff
@@ -1,51 +1,56 @@
-import { Octokit } from 'octokit';
-import projectsConfig from '../../projects.config';
+import { Octokit } from "octokit";
+import projectsConfig from "../../projects.config";
+import { cache } from "../../utils/cache";
 
 export default async function handler(req, res) {
-    const { repo, vercel } = req.query;
-    const owner = projectsConfig.organization;
-
-    const octokit = new Octokit({
-        auth: process.env.GITHUB_TOKEN
-    });
-
-    const branches = await octokit.request('GET /repos/{owner}/{repo}/branches', {
-        owner,
-        repo,
-        protected: false
-    });
-
-    const slotsData = await Promise.all(
-        branches.data.map(async ({ name, commit: lastCommit }) => {
-            const commit = await octokit.request('GET /repos/{owner}/{repo}/commits/{ref}', {
-                owner,
-                repo,
-                ref: lastCommit.sha
-            });
-
-            const slotSuffix = name.replace(/\//g, '-');
-            const slotUrl = `https://${vercel}-${slotSuffix}.vercel.app`;
-            return {
-                branch: slotSuffix,
-                slotUrl,
-                date: commit.data.commit.committer.date,
-                commitUrl: commit.data.html_url,
-                commitMessage: commit.data.commit.message,
-                commitAuthor: commit.data.committer?.login
-            };
-        })
-    );
-
-    const sortedSlots = slotsData.sort((a, b) => {
-        if (a.date > b.date) {
-            return -1;
+  const { repo, vercel } = req.query;
+  const owner = projectsConfig.organization;
+
+  const octokit = new Octokit({
+    auth: process.env.GITHUB_TOKEN,
+  });
+
+  const branches = await octokit.request("GET /repos/{owner}/{repo}/branches", {
+    owner,
+    repo,
+    protected: false,
+  });
+
+  const slotsData = await Promise.all(
+    branches.data.map(async ({ name, commit: lastCommit }) => {
+      const commit = await octokit.request(
+        "GET /repos/{owner}/{repo}/commits/{ref}",
+        {
+          owner,
+          repo,
+          ref: lastCommit.sha,
         }
-        if (a.date < b.date) {
-            return 1;
-        }
-
-        return 0;
-    });
-
-    res.status(200).json(sortedSlots);
+      );
+
+      const slotSuffix = name.replace(/\//g, "-");
+      const slotUrl = `https://${vercel}-${slotSuffix}.vercel.app`;
+      return {
+        branch: slotSuffix,
+        slotUrl,
+        date: commit.data.commit.committer.date,
+        commitUrl: commit.data.html_url,
+        commitMessage: commit.data.commit.message,
+        commitAuthor: commit.data.committer?.login,
+      };
+    })
+  );
+
+  const sortedSlots = slotsData.sort((a, b) => {
+    if (a.date > b.date) {
+      return -1;
+    }
+    if (a.date < b.date) {
+      return 1;
+    }
+
+    return 0;
+  });
+
+  cache(res, 60 * 60 * 10);
```

Comment:
Caching not working in Vercel Serverless without state. We run function once. We can control this only by [Cache-Control](https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching#recommended-cache-control) header.